src/m365/spo/commands/tenant/tenant-applicationcustomizer-add.ts

```diff
@@ -0,0 +1,189 @@
+      }
+      else if (!solution.SkipFeatureDeployment) {
+        throw 'The solution has not been deployed to all sites. Make sure to deploy this solution to all sites.';
+      }
+      await this.addTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getAppCatalogUrl(): Promise<string> {
+    const spoTenantAppCatalogUrlGetCommandOutput: CommandOutput = await Cli.executeCommandWithOutput(spoTenantAppCatalogUrlGetCommand as Command, { options: { output: 'text', _: [] } });
+    const appCatalogUrl: string | undefined = spoTenantAppCatalogUrlGetCommandOutput.stdout;
+    if (!appCatalogUrl) {
+      throw 'Cannot add tenant-wide application customizer as app catalog cannot be found';
+    }
+    return appCatalogUrl;
+  }
+
+  private async getSolutionIdFromComponentManifestList(appCatalogUrl: string, clientSideComponentId: string, logger: Logger): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr('Retrieving component manifest item from the ComponentManifests list on the app catalog site so that we get the solution id');
+    }
+
+    const camlQuery = `<View><ViewFields><FieldRef Name='ClientComponentId'></FieldRef><FieldRef Name='SolutionId'></FieldRef></ViewFields><Query><Where><Eq><FieldRef Name='ClientComponentId' /><Value Type='Guid'>${clientSideComponentId}</Value></Eq></Where></Query></View>`;
+    const commandOptions: spoListItemListCommandOptions = {
+      webUrl: appCatalogUrl,
+      listUrl: `${urlUtil.getServerRelativeSiteUrl(appCatalogUrl)}/Lists/ComponentManifests`,
+      camlQuery: camlQuery,
+      verbose: this.verbose,
+      debug: this.debug,
+      output: 'json'
+    };
+    const output = await Cli.executeCommandWithOutput(spoListItemListCommand as Command, { options: { ...commandOptions, _: [] } });
+    const outputParsed = JSON.parse(output.stdout);
+    if (!outputParsed.length) {
+      throw 'No component found with the specified clientSideComponentId found in the component manifest list. Make sure that the application is added to the application catalog';
+    }
+    return outputParsed[0].SolutionId;
+  }
+
+  private async getSolutionFromAppCatalog(appCatalogUrl: string, solutionId: string, logger: Logger): Promise<Solution> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving solution with id ${solutionId} from the application catalog`);
+    }
+
+    const camlQuery = `<View><ViewFields><FieldRef Name='SkipFeatureDeployment'></FieldRef><FieldRef Name='ContainsTenantWideExtension'></FieldRef></ViewFields><Query><Where><Eq><FieldRef Name='AppProductID' /><Value Type='Guid'>${solutionId}</Value></Eq></Where></Query></View>`;
+    const commandOptions: spoListItemListCommandOptions = {
+      webUrl: appCatalogUrl,
+      listUrl: `${urlUtil.getServerRelativeSiteUrl(appCatalogUrl)}/AppCatalog`,
```

Comment:
Could you add the fields option and determine what you want to retrieve. otherwise, you'll get all properties in the HTTP response, which is a waste of resources.

Reply:
@martinlingstuyl  I specify the fields which I want to retrieve in my camlquery by using the `<ViewFields>`, which makes the option `fields` redundant.